core/src/main/scala/me/jeffmay/neo4j/client/cypher/scalacheck/CypherStatementGenerators.scala

```diff
@@ -0,0 +1,48 @@
+package me.jeffmay.neo4j.client.cypher.scalacheck
+
+import me.jeffmay.neo4j.client.cypher._
+import org.scalacheck.{Shrink, Gen}
+
+object CypherStatementGenerators extends CypherStatementGenerators
+trait CypherStatementGenerators extends CypherValueGenerators {
+
+  /**
+    * Generates a valid field name where the first character is alphabetical and the remaining chars
+    * are alphanumeric.
+    */
+  def genFieldName: Gen[String] = Gen.identifier
+
+  def genFields: Gen[(String, CypherValue)] = Gen.zip(genFieldName, genCypherValue)
+
+  /**
+    * Generates a nested array at the specified depth and width.
+    */
+  lazy val genCypherProps: Gen[CypherProps] = {
+    for {
+      fields <- Gen.listOf(genFields)
+    } yield Map(fields: _*)
+  }
+
+  /**
+    * Generates the parameters of a [[CypherStatement]].
+    */
+  lazy val genCypherParams: Gen[Map[String, CypherProps]] = {
+    Gen.mapOf(Gen.zip(
+      Gen.identifier,
+      Gen.mapOf(Gen.zip(
+        Gen.identifier,
+        genCypherValue
+      ))
+    ))
+  }
+
+  implicit val shrinkCypherProps: Shrink[CypherProps] = Shrink { params =>
+    implicit val shrinkIdentifier = Shrink[String](id => Shrink.shrink[String](id).filterNot(_.isEmpty))
+    Shrink.shrinkContainer2[Map, String, CypherValue].shrink(params)
+  }
+
+  implicit val shrinkCypherParams: Shrink[Map[String, CypherProps]] = Shrink { params =>
+    implicit val shrinkIdentifier = Shrink[String](id => Shrink.shrink[String](id).filterNot(_.isEmpty))
+    Shrink.shrinkContainer2[Map, String, CypherProps].shrink(params)
+  }
```

Comment:
I should document this, but empty string keys are not allowed for params or props, so this prevents the `Shrink`s from creating invalid input. I wrote these to debug the issue but it turns out that the issue was with an empty map altogether, which is allowed. I'm not sure how helpful these are, but I should at least document them.